File: src/kakao/demotest/solution2.py
# https://www.welcomekakao.com/tryouts/1467/intro
def solution(arr):
    table = [0] * (len(arr)+1)
    for i in arr:
        if i > len(arr) or i < 1 or table[i]!=0:
            return False
        table[i] = i
    return True

# print(solution([1,2,3]))
# print(solution([1,2,3,3]))
# print(solution([4,1,3]))

File: src/kakao/demotest/solution2.js
// https://www.welcomekakao.com/tryouts/1467/intro
const solution = function(arr) {

  let table = []
  for(let i=0; i<=arr.length; i++){
    table[i] = 0;
  }
  for(let i of arr){
    if(i>arr.length || table[i]!=0){
      return false;
    }
    table[i] = i;
  }
  return true;
}

// console.log(solution([1, 2, 3]));
// console.log(solution([3, 1, 2, 4]));
// console.log(solution([4, 2, 3]));
